src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1233,144 @@ private int SkipBlockComment(int i)
             return i;
         }
 
-        private static char Backtick(char c)
+        private string Backtick(char c)
         {
             switch (c)
             {
-                case '0': return '\0';
-                case 'a': return '\a';
-                case 'b': return '\b';
-                case 'e': return '\u001b';
-                case 'f': return '\f';
-                case 'n': return '\n';
-                case 'r': return '\r';
-                case 't': return '\t';
-                case 'v': return '\v';
-                default: return c;
+                case '0': return "\0";
+                case 'a': return "\a";
+                case 'b': return "\b";
+                case 'e': return "\u001b";
+                case 'f': return "\f";
+                case 'n': return "\n";
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
             }
         }
 
+        private string ScanUnicodeEscapeSequence()
+        {
+            int escSeqStartIndex = _currentIndex - 2;
+            bool isBracketedSequence = false;
+            bool isTerminated = false;
+
+            char c = PeekChar();
+            if (c == '{')
+            {
+                SkipChar();
+                isBracketedSequence = true;
+            }
+
+            int maxNumberOfHexDigits = (isBracketedSequence ? 6 : 4);
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            int i;
+            for (i = 0; i < maxNumberOfHexDigits; i++)
```

Comment:
I believe we should follow the resultant [comment](https://github.com/PowerShell/PowerShell/issues/2751#issuecomment-306937602) from @mklement0. Based on this my understanding is:
1. We should support both forms: `u{...} and `u... . The first is preffered and we should recommend it in our documentation.
2. We should support variable-digit-count for both form. 
    It is more "native" for script shells and more user-frendly.
    It remove "tons" of errors in reporting. Current messages is unduly detailed and looks as noisy. I guess we'd have only two errors (1) - '}' absent and (2) Invalid Unicode escape sequence: must be `u{<hex-digits>} or `u<hex-digits>, using between 1 and 6 hex digits, up to a maximum of 10FFFF.
    We should generate "maximum is 10FFFF" in convertion time (Char.ConvertFromUtf32) and save CoreFX exception in InnerException. In this way we will fully follow the CoreFX support for Unicode chars.
    It's all make our code easier and UX better